src/getMovieReviews.js

```diff
@@ -0,0 +1,69 @@
+import fetch from "node-fetch";
+
+//fetch all reviews for a movie when there are more reviews than in the first fetch
+async function xloadMovieReviews(id, number) {
+    const API_BASE = "https://plankton-app-xhkom.ondigitalocean.app/api";
+    const Q_MOVIE = "filters[movie]=";
+    const Q_PSIZE = "pagination[pageSize]=";
+
+    const response = await fetch(API_BASE + "/reviews" + "?" + Q_MOVIE + id + "&" + Q_PSIZE + number);
+    const payload = await response.json();
+    return payload;
+};
+
+//test if there are more reviews to get & call for the fetch 
+async function evaluateIndex(id, index, number, payload) {
+    if (index > 1) {
+        const reviews = await xloadMovieReviews(id, number);
+        return reviews;
+    } else {
+        return payload;
+    };
+};
+
+//calculate index for test if there are more reviews to get
+async function allReviews(id, payload) {
+    if (payload.meta) {
+        let xPageSize = payload.meta.pagination.pageSize;
+        let xTotalReviews = payload.meta.pagination.total;
+        let loadIndex = xTotalReviews / xPageSize;
+        const reviews = await evaluateIndex(id, loadIndex, xTotalReviews, payload);
+        return reviews;
+    } else {
+        return payload;
+    };
+};
+
+//prepare reviews for right content & format
+async function prepareReviews(reviews) {
+    const preparedReviews = reviews.data.map((obj) => {
+        if (obj.attributes.comment === null) {
+            obj.attributes.comment = "No comment";
```

Comment:
This feels to me like presentation logic, i.e. code that belongs in the "presentation layer" of the software. By that I mean it would have been better suited to go in the template or frontend JS (depending on how it ends up being used).

Having the code all the way back here in the internal logic of the backend means two things:

* Text decisions are being made in the backend. If someone wants to change (or translate) these labels, they will need to make changes to backend code.
* If the frontend wants to know whether a review has no comment, e.g. in order to style it differently, it will need to check whether the text is "No comment", which is more difficult, and more brittle, than checking if it's `null`.